runtime-modules/council/src/lib.rs

```diff
@@ -671,15 +722,15 @@ impl<T: Trait> EnsureChecks<T> {
             return Err(Error::StakeStillNeeded);
         }
 
-        if &candidate.account_id != account_id {
-            return Err(Error::InvalidAccountToUnstake);
-        }
+        // prevent membership and account combination change
+        Self::prevent_ids_mismatch(&candidate, account_id)?;
```

Comment:
I am not sure exactly what this test is trying to do, but it has the same problem as the other comment, namling comparing membership controller accounts and staking accounts.

Reply:
This function is no longer needed. Due to changes mentioned in the previous review-topic above.